src/dune_rules/dune_env.ml

```diff
@@ -232,8 +233,9 @@ module Stanza = struct
     enter
       (let+ pat =
          keyword "_" >>> return Any
-         <|> let+ p = Profile.decode in
-             Profile p
+         <|>
```

Comment:
IMO the old code looked better here.

Reply:
What do you think of the similar change in cmd code, where the let-expression is much bigger and the operators are not chained ?

Reply:
The old version is better there too. The additional line only pays off when the let expression has some ugly wrapping due to the lack of horizontal space. There was no such issue there.

Reply:
Can ocamlformat pick the one that causes the least wrapping? (at least under the hood)

Reply:
Not really, OCamlformat cannot compare two possible formatting. Only the input can be considered (though, multiple iterations on the whole code can be made).
But it can format differently whether there would be no wrapping at all and it can detect chained operators.

Such formatting could be possible:

```
<|> let+ p = Profile.decode in Profile p
<|> ...
```

falling back to this as soon as it doesn't fit on the line:

```
<|>
let+ p = Profile.decode in
(* longer code *)
Profile p
<|> ...
```

I'm mentioning chained operators because to me, the cmd code looks better with less indentation.
To me, that indentation *can* cause ugly wrapping means that it *does*.

Reply:
By looking at the diff again, I find that the reduced indentation do save lines by removing some wrapping. I also do not find a lot of chained operators so my previous suggestion is not valid.

Though, it should be possible have this formatting, as long as it fits on one line:
```ocaml
...
<|> let+ p = Profile.decode in Profile p
```

Should I add this before the next release ?

Reply:
not a huge fan of saving lines at all cost, I think that the old version was better.

Reply:
I went back and found the reason for this change: https://github.com/ocaml-ppx/ocamlformat/pull/2315
It's a consistency change, in the previous version, the line was broken on regular lets but not letops:
```ocaml
let _ =
  Cmd.v info
  @@ let+ trimmed_size = () in
     ()

let _ =
  ()
  <|> let+ p = Profile.decode in
      Profile p

let _ =
  Cmd.v info
  @@
  let trimmed_size = () in
  ()

let _ =
  ()
  <|>
  let p = Profile.decode in
  Profile p
```

The other form exists in the code: `bin/describe/describe_workspace.ml` line 378.

This diff can go away by keeping the line break from the source code, whether it's a let or a letop. (https://github.com/ocaml-ppx/ocamlformat/pull/2396)
We try to add as few of these as possible but the regression is large enough to justify it.

I'll merge this patch and update this preview PR after the weekend.